### PR TITLE
feat(sources/community/supabase): add Supabase community source

### LINKED ISSUE #442: design: add Supabase community source
## Summary

Add a new community `supabase` source that lets Coral query Supabase platform metadata through SQL. For users, this would make organizations, projects, edge functions, secrets, storage buckets, service health, database backups, branches, and PostgREST configuration queryable in a way that supports infrastructure visibility, project auditing, and DevOps workflows, while staying within Coral's current HTTP source-spec model and keeping the first version intentionally read-only.

## Motivation

Supabase is one of the most widely used open-source Firebase alternatives, powering backends for thousands of web and mobile applications. Teams building on Supabase often need to answer questions like:

- "What projects do we have and what's their health status?"
- "Which projects are inactive or paused?"
- "What edge functions are deployed to a project and are they using JWT verification?"
- "What secrets are configured for a project?"
- "Who are the members of our organization and do they have MFA enabled?"
- "Are all services (auth, realtime, storage, postgrest) healthy?"
- "What's the backup status for our production database?"
- "What database branches exist and which ones are tied to open PRs?"

Without a Coral source, users who want Supabase-aware agent workflows must write their own source spec. This source would make common platform-engineering, infrastructure-audit, and project-management workflows easy to query with SQL, and would naturally complement existing Coral sources like GitHub and Vercel.

## Proposed design

This would be a new community source under `sources/community/supabase`, implemented as a manifest-only HTTP source.

Inputs:

- `SUPABASE_ACCESS_TOKEN`
  - required secret
  - hint: Personal access token (PAT) from https://supabase.com/dashboard/account/tokens

Auth:

- HeaderAuth
- `Authorization: Bearer {{input.SUPABASE_ACCESS_TOKEN}}`

Proposed v1 tables:

- `supabase.organizations`
  - `GET /v1/organizations`
  - lists all organizations accessible to the authenticated user
  - key columns: id, name, slug

- `supabase.projects`
  - `GET /v1/projects`
  - lists all projects across all organizations
  - key columns: id, ref, name, region, status, created_at, organization_id, organization_slug, database__host, database__version, database__postgres_engine, database__release_channel

- `supabase.organization_members`
  - `GET /v1/organizations/{slug}/members`
  - requires `slug` filter
  - key columns: user_id, user_name, email, role_name, mfa_enabled

- `supabase.edge_functions`
  - `GET /v1/projects/{ref}/functions`
  - requires `project_ref` filter
  - key columns: id, slug, name, status, version, verify_jwt, import_map, entrypoint_path, created_at, updated_at

- `supabase.secrets`
  - `GET /v1/projects/{ref}/secrets`
  - requires `project_ref` filter
  - key columns: name, value (redacted by API), updated_at

- `supabase.storage_buckets`
  - `GET /v1/projects/{ref}/storage/buckets`
  - requires `project_ref` filter
  - key columns: id, name, owner, public, created_at, updated_at

- `supabase.service_health`
  - `GET /v1/projects/{ref}/health`
  - requires `project_ref` and `services` filters
  - key columns: name, healthy, status, error, info__name, info__version, info__description

- `supabase.backups`
  - `GET /v1/projects/{ref}/database/backups`
  - requires `project_ref` filter
  - rows extracted from the `backups` key in the response (top-level also contains PITR metadata)
  - key columns: id, status, is_physical_backup, inserted_at

- `supabase.branches`
  - `GET /v1/projects/{ref}/branches`
  - requires `project_ref` filter
  - requires a paid plan with branching enabled
  - key columns: id, name, project_ref, parent_project_ref, is_default, git_branch, pr_number, persistent, status, created_at, updated_at

- `supabase.postgrest_config`
  - `GET /v1/projects/{ref}/postgrest`
  - requires `project_ref` filter
  - single-object response using `row_strategy: direct`
  - key columns: db_schema, max_rows, db_extra_search_path, db_pool

Design details:

- use Supabase Personal Access Token (PAT) auth via bearer token
- no pagination needed — Management API list endpoints return all items in a single response
- top-level tables (`organizations`, `projects`) serve as discovery tables whose `slug` and `ref` columns are required filters for all other tables
- flatten commonly queried nested fields with double-underscore naming (`database__host`, `info__version`)
- keep the source read-only
- use `rows_path: [backups]` for the backups table to extract rows from the response wrapper
- use `row_strategy: direct` for `postgrest_config` since the endpoint returns a single config object
- virtual `from_filter` columns echo filter values back into result rows
- include Supabase rate-limit headers: `x-ratelimit-remaining`, `x-ratelimit-reset`

Affected surface:

- community source only
- expected files:
  - `sources/community/supabase/manifest.yaml`
  - `sources/community/supabase/README.md`

No CLI, config, MCP, or runtime surface changes are proposed.

## Drawbacks and trade-offs

This source would be intentionally narrow in v1. It would not include Postgres database SQL execution (querying the user's application data), write operations (project creation, pausing/resuming), network restrictions, auth configuration (providers, templates, hooks), custom domains, database migrations, or realtime inspection endpoints.

Most Supabase Management API list endpoints return all items in a single response without pagination. Accounts with very large numbers of projects or organizations may hit API-side limits, but this is uncommon in practice.

Secret values are redacted by the API — only key names and timestamps are visible through the Management API.

The `branches` table requires a paid Supabase plan with branching enabled; projects without branching will return an error.

## Alternatives considered

A broader Supabase source could include auth configuration, custom domains, database migrations, realtime channels, and storage object listing, but that expands scope too quickly for v1. A project-only source would be simpler but loses the cross-table value of joining organizations, members, functions, health, and backups. A focused read-only platform-metadata source is the better starting point.

## Open questions

- Does the ten-table scope feel right for v1, or should it start smaller?
- Should `postgrest_config` be included or deferred to a later version?
- Any columns that should be added or dropped?
- Happy to adjust based on feedback.

- [x] I've read the [What we welcome](https://github.com/withcoral/coral/blob/main/CONTRIBUTING.md#what-we-welcome-right-now) and [Please discuss first](https://github.com/withcoral/coral/blob/main/CONTRIBUTING.md#please-discuss-first) sections of CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by Arnav1709**
## Summary

Adds a new community source for Supabase under `sources/community/supabase/`.

This source enables querying **Supabase Management API** (`api.supabase.com`) platform metadata through SQL — organizations, projects, edge functions, secrets inventory, storage buckets, service health, database backups, branches, and PostgREST configuration. The source is manifest-only, read-only, and uses Coral's existing HTTP source-spec model. No CLI, MCP, config, or runtime changes.

> **Scope:** This source covers the Management API only. It does **not** query the PostgREST data-plane (`<ref>.supabase.co`) or project Postgres databases. To query application data stored in Supabase Postgres tables, users should connect with a Postgres-compatible source or client.

Closes #442

## What changed

Added:

- `sources/community/supabase/manifest.yaml`
- `sources/community/supabase/README.md`

## Tables

| Table | Endpoint | Notes |
|---|---|---|
| `organizations` | `GET /v1/organizations` | All orgs for the authenticated user |
| `projects` | `GET /v1/projects` | All projects across all orgs |
| `organization_members` | `GET /v1/organizations/{slug}/members` | Requires `slug` filter |
| `edge_functions` | `GET /v1/projects/{ref}/functions` | Requires `project_ref` filter |
| `secrets` | `GET /v1/projects/{ref}/secrets` | Requires `project_ref` filter; **names and timestamps only — values never returned** |
| `storage_buckets` | `GET /v1/projects/{ref}/storage/buckets` | Requires `project_ref` filter |
| `service_health` | `GET /v1/projects/{ref}/health` | Requires `project_ref`, `services` filters |
| `backups` | `GET /v1/projects/{ref}/database/backups` | Requires `project_ref` filter; rows extracted from `backups` key |
| `branches` | `GET /v1/projects/{ref}/branches` | Requires `project_ref` filter; needs branching plan |
| `postgrest_config` | `GET /v1/projects/{ref}/postgrest` | Requires `project_ref` filter; single-row `direct` strategy |

## Auth

Bearer token via `Authorization: Bearer <SUPABASE_ACCESS_TOKEN>`.

Inputs:

- `SUPABASE_ACCESS_TOKEN` — required secret, created at https://supabase.com/dashboard/account/tokens

### Least-privilege permissions

PATs carry the same privileges as the user account. For least-privilege access, Supabase supports fine-grained permissions. The minimum scopes required by this source are:

| Permission scope | Access level | Used by tables |
|---|---|---|
| `organizations_read` | Read | `organizations` |
| `members_read` | Read | `organization_members` |
| `projects_read` | Read | `projects` |
| `edge_functions_read` | Read | `edge_functions` |
| `edge_functions_secrets_read` | Read | `secrets` |
| `storage_read` | Read | `storage_buckets` |
| `project_admin_read` | Read | `service_health` |
| `backups_read` | Read | `backups` |
| `branching_production_read` or `branching_development_read` | Read | `branches` |
| `data_api_config_read` | Read | `postgrest_config` |

## Security notes

- **Secrets table is inventory-only.** The `secrets` table returns secret names and timestamps but **never exposes secret values**. The `value` column has been intentionally omitted.
- **Token scope.** PATs inherit full account privileges. The README documents fine-grained permission scopes for least-privilege access.
- **No credential surfaces.** This source does not expose database passwords, connection strings, or API keys from any table.

## Implementation notes

- All endpoints are under `https://api.supabase.com/v1/` and return JSON arrays directly (no pagination needed — the Management API returns all items in a single response)
- Rate-limit headers (`x-ratelimit-remaining`, `x-ratelimit-reset`) are declared
- Top-level tables (`organizations`, `projects`) serve as discovery tables — their `slug` and `ref` columns are the required filters for all other lookup-style tables
- Nested database fields are flattened with double-underscore naming (`database__host`, `database__version`, `database__postgres_engine`, `database__release_channel`)
- Nested service health info fields are flattened (`info__name`, `info__version`, `info__description`)
- `backups` response uses `rows_path: [backups]` to extract backup rows from the top-level response object (which also contains PITR metadata)
- `postgrest_config` uses `row_strategy: direct` because the endpoint returns a single config object, not an array
- Virtual `from_filter` columns echo filter values back into result rows for traceability

## Validation

- `coral source lint ./sources/community/supabase/manifest.yaml` — passes
- `make lint-sources` (`ryl`) — passes with no warnings
- **Live query validation** — successfully completed against a real Supabase account. `coral source test` passes for all 10 tables, and structural mapping handles edge cases gracefully (such as bubbling up 400/404s for paused/inactive projects).

## User-visible impact

New `supabase` source installable via:

```bash
coral source add supabase
```

### First-success path

```sql
-- 1. Verify credentials and discover organizations
SELECT id, name, slug FROM supabase.organizations;

-- 2. List all projects with status and region
SELECT id, ref, name, region, status, created_at
FROM supabase.projects;

-- 3. Drill into a project using the ref from step 2
SELECT id, name, slug, status, version, verify_jwt
FROM supabase.edge_functions
WHERE project_ref = 'abcdefghijklmnopqrst';
```

### More examples

```sql
-- Check which projects are inactive
SELECT ref, name, status, created_at
FROM supabase.projects
WHERE status = 'INACTIVE';

-- List members of an organization
SELECT user_name, email, role_name, mfa_enabled
FROM supabase.organization_members
WHERE slug = 'my-org-slug';

-- Secret inventory (names and timestamps only)
SELECT name, updated_at
FROM supabase.secrets
WHERE project_ref = 'abcdefghijklmnopqrst';

-- Check service health
SELECT name, healthy, status, info__version, error
FROM supabase.service_health
WHERE project_ref = 'abcdefghijklmnopqrst'
  AND services = 'auth,realtime,storage,rest';

-- List database backups
SELECT id, status, is_physical_backup, inserted_at
FROM supabase.backups
WHERE project_ref = 'abcdefghijklmnopqrst';

-- Check PostgREST configuration (Data API settings, not application data)
SELECT db_schema, max_rows, db_pool
FROM supabase.postgrest_config
WHERE project_ref = 'abcdefghijklmnopqrst';
```

## Known limitations

- **Read-only** — no create, update, or delete operations
- **Management API only** — this source queries platform metadata at `api.supabase.com`, not project Postgres databases; to query application data, use a Postgres-compatible source pointed at `<ref>.supabase.co`
- **No pagination** — the Management API returns all items in a single response; accounts with very large project/org counts may hit API-side limits
- **Branching** — the `branches` table requires a paid plan with branching enabled on the project
- **Service health** — requires specifying which services to check via the `services` filter (e.g. `auth,realtime,storage,rest`)
- **Inactive projects** — for paused/inactive projects, querying `service_health` returns an API 400 and `postgrest_config` returns an API 404; Coral surfaces these upstream errors rather than crashing

## Out of scope

Not included in v1:

- Postgres database SQL execution (querying the user's application data via PostgREST data-plane)
- Write operations (project creation, pausing/resuming, deployments)
- Network restrictions and SSL enforcement configuration
- Auth configuration (providers, templates, hooks)
- Custom domains and vanity subdomains
- Database migrations and type generation
- Realtime inspection endpoints

### CODE REVIEW

**Comment by Arnav1709** [comment]
@antonmry can you please review this?

**Comment by antonmry** [comment]
@Arnav1709 there's already a reviewer assigned to this PR, let's wait for his review

**Comment by Arnav1709** [comment]
hey @iaptsiauri can you please review this pr!!

**Review by jsummerfield — COMMENTED** [review]
Source review finding:

- Blocking: this PR does not currently have the `cla signed` label. The repo contribution rules require the CLA before a community source can be approved. Please complete the CLA flow and get the label applied, then this can receive a source-spec content review.

**Review by jsummerfield — COMMENTED** [review]
Findings:

- High: this branch currently deletes 21 unrelated files under `sources/community` when compared with current `main`. A community-source PR should add or update its own source directory without removing existing sources. Please merge/rebase current `main` and restore these unrelated files before this can be approved. Examples from the diff:

- `sources/community/beehiiv/README.md`
- `sources/community/beehiiv/manifest.yaml`
- `sources/community/cal/README.md`
- `sources/community/cal/manifest.yaml`
- `sources/community/clickhouse/README.md`
- `sources/community/clickhouse/manifest.yaml`
- `sources/community/cloudflare/README.md`
- `sources/community/cloudflare/manifest.yaml`

**Comment by jsummerfield** [comment]
Source review finding:

The secrets table needs one more pass before approval. The source exposes `supabase.secrets.value` (`sources/community/supabase/manifest.yaml:347`), but both the manifest guide and README say the API returns empty/redacted values (`sources/community/supabase/manifest.yaml:326`, `sources/community/supabase/README.md:25`, `sources/community/supabase/README.md:117`). Supabase's Management API reference for `GET /v1/projects/{ref}/secrets` documents the response with `name`, `value`, and `updated_at`, and the newer docs list the required fine-grained permission as `edge_functions_secrets_read`: https://supabase.com/docs/reference/api/v1-check-vanity-subdomain-availability

Could you either remove the `value` column for v1, or keep it and make the README/manifest guide explicit that this table may expose secret values to anyone who can query the installed Coral source? My preference is to omit `value` unless there is a clear user workflow that needs it; secret-name inventory is already useful and much safer by default.

File: sources/community/supabase/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+          kind: from_filter
+          key: project_ref
+
+  # ──────────────────────────────────────────────
+  # Secrets
+  # ──────────────────────────────────────────────
+  - name: secrets
+    description: >
+      Project secrets (environment variables available to Edge Functions).
+      Returns secret names and their last-updated timestamps for inventory purposes.
```
[the remark this answers is not in the document]
Comment (Arnav1709) [reply sources/community/supabase/manifest.yaml]: @cubic-dev-ai retry

**Comment by Arnav1709** [comment]
> Source review finding:
> 
> The secrets table needs one more pass before approval. The source exposes `supabase.secrets.value` (`sources/community/supabase/manifest.yaml:347`), but both the manifest guide and README say the API returns empty/redacted values (`sources/community/supabase/manifest.yaml:326`, `sources/community/supabase/README.md:25`, `sources/community/supabase/README.md:117`). Supabase's Management API reference for `GET /v1/projects/{ref}/secrets` documents the response with `name`, `value`, and `updated_at`, and the newer docs list the required fine-grained permission as `edge_functions_secrets_read`: https://supabase.com/docs/reference/api/v1-check-vanity-subdomain-availability
> 
> Could you either remove the `value` column for v1, or keep it and make the README/manifest guide explicit that this table may expose secret values to anyone who can query the installed Coral source? My preference is to omit `value` unless there is a clear user workflow that needs it; secret-name inventory is already useful and much safer by default.

I've removed the value column from the supabase.secrets table entirely, so it now acts strictly as an inventory of secret names and timestamps. I've also updated the README and the manifest guide to clarify this change and removed mentions of redacted values. Thanks for the suggestion!

**Comment by jsummerfield** [comment]
As we consolidate duplicate provider submissions into this PR, here are a few product-quality points we should preserve before merge:

- Keep the scope crisp: say whether this source is Supabase Management API, PostgREST project data-plane, or both.
- Make token type and least-privilege permissions explicit, especially around organization/project metadata and edge function secrets.
- Prefer inventory-safe defaults for sensitive surfaces such as secrets; avoid returning secret values unless the README clearly explains the risk and need.
- Keep a first-success path with organization/projects/source test examples.
- If PostgREST data-plane workflows are out of scope, say where users should expect to query project tables.

File: sources/community/supabase/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+| Health — Read                | Read         | `service_health`                                    |
+
+For third-party integrations, Supabase also supports OAuth2 with scoped tokens.
+See [Build a Supabase Integration](https://supabase.com/docs/guides/integrations/build-a-supabase-integration)
+for details.
+
+### 2. Add the source
+
+```bash
+coral source add supabase
```
[the remark this answers is not in the document]
Comment (Arnav1709) [reply sources/community/supabase/README.md]: @cubic-dev-ai recheck

**Comment by Arnav1709** [comment]
> As we consolidate duplicate provider submissions into this PR, here are a few product-quality points we should preserve before merge:
> 
> * Keep the scope crisp: say whether this source is Supabase Management API, PostgREST project data-plane, or both.
> * Make token type and least-privilege permissions explicit, especially around organization/project metadata and edge function secrets.
> * Prefer inventory-safe defaults for sensitive surfaces such as secrets; avoid returning secret values unless the README clearly explains the risk and need.
> * Keep a first-success path with organization/projects/source test examples.
> * If PostgREST data-plane workflows are out of scope, say where users should expect to query project tables.

All five points addressed in this update:

1. **Scope clarity** — Both the README and manifest description now explicitly state this is the **Supabase Management API** (`api.supabase.com`) only. The README opens with a blockquote callout distinguishing it from the PostgREST data-plane at `<ref>.supabase.co`.

2. **Token type and least-privilege permissions** — Added a permissions table to the README listing the minimum fine-grained scopes (Organizations Read, Projects Read, Edge Functions Read, Edge Function Secrets Read, Health Read). The manifest input hint also now references fine-grained permissions.

3. **Inventory-safe secrets** — The `value` column was removed in the previous commit. This update reinforces the messaging: the secrets table description, guide, and a new "Security notes" section in the README all explicitly state that values are **never returned**.

4. **First-success path** — New "Quick start" section in the README walks users through: verify credentials with `organizations` → discover `ref` values with `projects` → drill into project-scoped tables.

5. **PostgREST data-plane out of scope** — Addressed in the scope callout at the top of the README, reiterated in the Limitations section, and the out-of-scope list now calls out "Postgres database SQL execution (querying the user's application data via PostgREST data-plane)" with guidance to use a Postgres-compatible source instead.

**Review by jsummerfield — CHANGES_REQUESTED** [review]
Findings:

- High: the least-privilege permission list in `sources/community/supabase/README.md:27-33` is still too broad for the fine-grained Supabase PAT setup. Several tables require specific permissions such as `members_read`, `storage_read`, `backups_read`, `branching_production_read` or `branching_development_read`, `data_api_config_read`, and `project_admin_read`, not just the grouped “Organizations / Projects / Health” wording. Please make the README permissions table match the endpoint-level requirements from the Supabase Management API spec: https://api.supabase.com/api/v1-json

- High: the `service_health` examples and guide use `services = 'auth,realtime,storage,postgrest'` (`sources/community/supabase/manifest.yaml:439-455`, `sources/community/supabase/README.md:105-109`, `:177-178`), but the Management API uses `rest`, not `postgrest`, for that service enum. A user following the quick start can hit a validation error here. Docs: https://supabase.com/docs/reference/api/v1-get-services-health

- Medium: the backup and branch status examples still need to match Supabase’s documented enums. `IN_PROGRESS` is not in the documented backup status enum (`COMPLETED`, `FAILED`, `PENDING`, `REMOVED`, `ARCHIVED`, `CANCELLED`), and `PAUSING` / `REMOVED` are project-style lifecycle values rather than the documented branch response status examples (`sources/community/supabase/manifest.yaml:527-550`, `:639-644`). Docs: https://supabase.com/docs/reference/api/v1-list-all-backups and https://supabase.com/docs/reference/api/v1-list-all-branches

The previous `secrets.value` concern is resolved: the table now exposes only name/timestamp metadata, which is the right safer default. Thanks for tightening that up.

**Comment by Arnav1709** [comment]
> Findings:
> 
> * High: the least-privilege permission list in `sources/community/supabase/README.md:27-33` is still too broad for the fine-grained Supabase PAT setup. Several tables require specific permissions such as `members_read`, `storage_read`, `backups_read`, `branching_production_read` or `branching_development_read`, `data_api_config_read`, and `project_admin_read`, not just the grouped “Organizations / Projects / Health” wording. Please make the README permissions table match the endpoint-level requirements from the Supabase Management API spec: https://api.supabase.com/api/v1-json
> * High: the `service_health` examples and guide use `services = 'auth,realtime,storage,postgrest'` (`sources/community/supabase/manifest.yaml:439-455`, `sources/community/supabase/README.md:105-109`, `:177-178`), but the Management API uses `rest`, not `postgrest`, for that service enum. A user following the quick start can hit a validation error here. Docs: https://supabase.com/docs/reference/api/v1-get-services-health
> * Medium: the backup and branch status examples still need to match Supabase’s documented enums. `IN_PROGRESS` is not in the documented backup status enum (`COMPLETED`, `FAILED`, `PENDING`, `REMOVED`, `ARCHIVED`, `CANCELLED`), and `PAUSING` / `REMOVED` are project-style lifecycle values rather than the documented branch response status examples (`sources/community/supabase/manifest.yaml:527-550`, `:639-644`). Docs: https://supabase.com/docs/reference/api/v1-list-all-backups and https://supabase.com/docs/reference/api/v1-list-all-branches
> 
> The previous `secrets.value` concern is resolved: the table now exposes only name/timestamp metadata, which is the right safer default. Thanks for tightening that up.

I've addressed all the feedback in the latest commit:
- **Permissions Table**: Updated `README.md` to map each table directly to its specific endpoint-level scope (e.g., `organizations_read`, `members_read`, `projects_read`, `edge_functions_read`, `edge_functions_secrets_read`, `storage_read`, `project_admin_read`, `backups_read`, `branching_production_read` / `branching_development_read`, and `data_api_config_read`).
- **Service Health Enum**: Changed `postgrest` to `rest` in both the `manifest.yaml` guide/descriptions and the `README.md` quick start and limitations examples.
- **Status Enums**: 
  - Updated the backup status description in `manifest.yaml` to use `PENDING` instead of `IN_PROGRESS`.
  - Updated the branch status description in `manifest.yaml` to use documented branch statuses (`CREATING_PROJECT`, `RUNNING_MIGRATIONS`, `MIGRATIONS_PASSED`, `FUNCTIONS_DEPLOYED`) instead of the project-level lifecycle values.
  
I have also updated the PR description to reflect these changes.

**Review by jsummerfield — CHANGES_REQUESTED** [review]
Findings

- High: `sources/community/supabase/manifest.yaml:327`-`337` and `sources/community/supabase/README.md:155`-`157` say the `secrets` table is inventory-only and secret values are never returned. The SQL columns omit `value`, which is good, but the upstream Supabase Management API response for `GET /v1/projects/{ref}/secrets` includes secret values. That means the source still fetches credential material and the current README overstates the safety boundary. Please either remove this table, or document explicitly that Supabase returns secret values upstream and this Coral source intentionally does not expose the `value` field in SQL results. Docs: https://api.supabase.com/api/v1

- High: `sources/community/supabase/manifest.yaml:685`-`704` exposes `postgrest_config`, while `sources/community/supabase/README.md:160` says the source has no credential surfaces. Supabase's Management API response for project PostgREST config includes sensitive `jwt_secret` material, even though this manifest does not expose that column. Please either drop the table or document the upstream secret-bearing response and the fact that Coral intentionally omits `jwt_secret`. Docs: https://api.supabase.com/api/v1

- Medium: `sources/community/supabase/manifest.yaml:447`-`455` models `service_health.services` as one comma-separated string, and the README examples use `services = 'auth,realtime,storage,rest'`. Supabase's OpenAPI schema defines `services` as a required array enum. Please verify the live query-string behavior; if the API expects repeated params or array encoding, expose a single-service query shape or otherwise model the provider-supported request instead of a comma-joined string. Docs: https://api.supabase.com/api/v1

Thanks for tightening the permissions table and removing direct secret columns. The remaining blockers are about being precise where the upstream API response itself can contain secrets, even when the Coral table intentionally drops those fields.

**Comment by Arnav1709** [comment]
> Findings
> 
> * High: `sources/community/supabase/manifest.yaml:327`-`337` and `sources/community/supabase/README.md:155`-`157` say the `secrets` table is inventory-only and secret values are never returned. The SQL columns omit `value`, which is good, but the upstream Supabase Management API response for `GET /v1/projects/{ref}/secrets` includes secret values. That means the source still fetches credential material and the current README overstates the safety boundary. Please either remove this table, or document explicitly that Supabase returns secret values upstream and this Coral source intentionally does not expose the `value` field in SQL results. Docs: https://api.supabase.com/api/v1
> * High: `sources/community/supabase/manifest.yaml:685`-`704` exposes `postgrest_config`, while `sources/community/supabase/README.md:160` says the source has no credential surfaces. Supabase's Management API response for project PostgREST config includes sensitive `jwt_secret` material, even though this manifest does not expose that column. Please either drop the table or document the upstream secret-bearing response and the fact that Coral intentionally omits `jwt_secret`. Docs: https://api.supabase.com/api/v1
> * Medium: `sources/community/supabase/manifest.yaml:447`-`455` models `service_health.services` as one comma-separated string, and the README examples use `services = 'auth,realtime,storage,rest'`. Supabase's OpenAPI schema defines `services` as a required array enum. Please verify the live query-string behavior; if the API expects repeated params or array encoding, expose a single-service query shape or otherwise model the provider-supported request instead of a comma-joined string. Docs: https://api.supabase.com/api/v1
> 
> Thanks for tightening the permissions table and removing direct secret columns. The remaining blockers are about being precise where the upstream API response itself can contain secrets, even when the Coral table intentionally drops those fields.

- `secrets`: documented that Supabase returns secret values upstream, but this Coral source intentionally omits the `value` field from SQL results.
- `postgrest_config`: documented that Supabase returns secret-bearing fields such as `jwt_secret` upstream, but this Coral source intentionally omits those fields from SQL results.
- `service_health`: changed the SQL-facing filter from comma-separated `services` to a single-service `service` filter, while still sending Supabase's provider query parameter as `services=<service>`.

**Review by jsummerfield — APPROVED** [review]
No blocking findings on the current head.

The previous blockers are resolved: `secrets` and `postgrest_config` now explicitly document that Supabase returns sensitive fields upstream while Coral intentionally omits them from SQL output, and `service_health` is modeled as a single-service query instead of a comma-joined array. The permissions table and source scope are now clear, local `coral source lint` passes, and GitHub source checks are green.

One non-blocking polish item for a follow-up or pre-merge cleanup: the `service_health` guide/README examples mention `auth`, `realtime`, `storage`, and `rest`, but Supabase's enum also includes `db`, `db_postgres_user`, `pooler`, and `pg_bouncer`. Listing the full set would make database/pooler health discovery easier.

Thanks for iterating this into a much safer Management API source.
